src/components/SubmitClassificationForm.jsx

```diff
@@ -70,9 +69,13 @@ class SubmitClassificationForm extends React.Component {
       this.context.googleLogger.logEvent({ type: 'complete-classification-and-talk' });
     }
 
-    this.props.dispatch(submitClassification())
+    if (this.props.currentSubject) {
+      const id = this.props.currentSubject.id;
```

Comment:
Minor safety catch needed here - it's possible for `this.props.currentSubject` to be `null`